### PR TITLE
Add apicurio namespace

### PR DESCRIPTION
**Pull request by paoloantinori**
Hi @durandom !

I've followed the docs to ask for the onboarding of Apicurio team, with the initial goal of deploying an instance of Apicurio Registry.
I'm honestly not sure if I've done things properly:
1. I had no clue which cluster I should have picked and based on what criteria, ended up with `dev`
2. I haven't performed this step: https://www.operate-first.cloud/users/support/docs/onboarding_to_cluster.md#adding-users-to-plaintext-overlay-patch since it was too obscure and I also had the suspect that the example above might have performed all operations already
3. I've even had funny errors with `toolbox` that somehow didn't see the part of filesystem where I keep my gh repos.

But other than that, I hope I've put most of the right things there.

Plase let me know if I have to fix or change anything

### CODE REVIEW

**Review by harshad16 — COMMENTED** [review]
The overlay dev is for dev work of operate-first.
cluster-scope/overlays/moc there are few instance in here
zero is the preferred on by most of user.

i will add others in this pr as well, who could guide with this better.
/hold

**Comment by 4n4nd** [comment]
@paoloantinori yeah the dev environment is for our development purposes, you want the moc/zero env. You can leave the list os users empty and I can just add an encrypted users list for you.

**Comment by paoloantinori** [comment]
@4n4nd thank you. Actually, I was curious to check the end-to-end process by myself. I've just updated the PR with the encrypted list of users, could you please confirm this is what it was expected?

thank you

File: cluster-scope/overlays/moc/zero/groups/apicurio.enc.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+apiVersion: user.openshift.io/v1
```
Comment (tumido) [inline cluster-scope/overlays/moc/zero/groups/apicurio.enc.yaml:1]: This group resource is encrypted correctly, however there are few changes needed:

1. Can you please relocate it to `cluster-scope/overlays/moc/common/groups/apicurio.enc.yaml`? The change is `s/zero/common/` in the path. It will essentially make the user group is available on all MOC clusters allowing you simpler migration and ArgoCD management. 
2. Additionally please include the resource in [`cluster-scope/overlays/moc/common/secret-generator.yaml`](https://github.com/operate-first/apps/blob/master/cluster-scope/overlays/moc/common/secret-generator.yaml).
3. Please change the usernames mentioned in `users` list of the user group from GitHub nicks to MOC SSO usernames - it's the email address you use whey you log into the web console (https://console-openshift-console.apps.zero.massopen.cloud/ ) using `moc-sso` oauth. We currently don't have GitHub auth enabled so the usernames you've listed here won't ever get assigned.

File: cluster-scope/overlays/moc/zero/secret-generator.yaml [hunk 2]
```diff
@@ -5,5 +5,5 @@ metadata:
   name: secret-generator
 files:
   - groups/cluster-admins.enc.yaml
-  - oauths/moc-sso-client-secret.enc.yaml
+  - oauths/apicurio.enc.yaml
```
Comment (tumido) [inline cluster-scope/overlays/moc/zero/secret-generator.yaml:8]: Please revert this change, instead list the encrypted resource along other user groups in the `moc/common` overlay as described above.

**Review by tumido — DISMISSED** [review]
Hi @paoloantinori ! Welcome! :slightly_smiling_face: I appreciate you want to serve the onboarding 100% yourself! It helps us greatly to understand the pain points users have and face with this. Thank you! :+1: 

There are few changes needed to the encrypted user group in order for it to work properly. Other than that the PR looks good!

**Comment by tumido** [comment]
btw, @paoloantinori I've just noticed: This PR might be a great inspiration for you on what the changes should look like in the overlays:
https://github.com/operate-first/apps/pull/478/

File: cluster-scope/overlays/moc/zero/secret-generator.yaml (hunk 2 above)
Comment (paoloantinori) [reply cluster-scope/overlays/moc/zero/secret-generator.yaml]: done

File: cluster-scope/overlays/moc/zero/groups/apicurio.enc.yaml (hunk 1 above)
Comment (paoloantinori) [reply cluster-scope/overlays/moc/zero/groups/apicurio.enc.yaml]: done

**Comment by paoloantinori** [comment]
@tumido thanks for your support. I've fixed the original PR and confirm to you that I've regenerated the encrypted file to use emails instead of gh usernames!

paolo

**Comment by tumido** [comment]
Much better! Thank you @paoloantinori ! It seems I forgot to mention 2 additional changes, apologies.

If you take a closer look at https://github.com/operate-first/apps/pull/478 you can see the group *base* has to be referenced as well.

That can be done by referencing it in `common/kustomization.yaml`:
https://github.com/operate-first/apps/blob/0185464941a953a62ac32a416c4680305da63cc7/cluster-scope/overlays/moc/common/kustomization.yaml#L8

And the target cluster has to be aware that it should deploy the namespace resource via listing it in `zero/kustomization.yaml` e.g.:
https://github.com/operate-first/apps/blob/0185464941a953a62ac32a416c4680305da63cc7/cluster-scope/overlays/moc/zero/kustomization.yaml#L35

To explain this a bit `overlays/moc/zero` and `overlays/moc/infra` both inherit from `overlays/moc/common`. The Zero overlay get's deployed to the [Zero cluster](console-openshift-console.apps.zero.massopen.cloud) - that's where you want your namespace to live at. On the other hand the Infra cluster is hosting our ArgoCD instance, hence you want to have your user group available on both clusters so you use the same context no matter which cluster you're accessing (therefore the group resource lives in the `overlays/moc/common` so it is pulled into both clusters) :slightly_smiling_face:

**Review by 4n4nd — COMMENTED** [review]
/lgtm

**Comment by paoloantinori** [comment]
Done thank you. 

Was this last step this line in the docs?

https://www.operate-first.cloud/users/support/docs/onboarding_to_cluster.md#enabling-namespace-deployment-to-a-specific-cluster

I have skipped that because I hadn't really understood what it was supposed to do and I thought it was there for future reference, when I was supposed to alter some config in the future.

**Comment by 4n4nd** [comment]
@paoloantinori because you added the namespace manifests to the kustomization in [moc/zero directory](https://github.com/operate-first/apps/blob/484ae6e797f5f1271ec8dc11ce277de5faadebf0/cluster-scope/overlays/moc/zero/kustomization.yaml), the namespace will be created in the zero cluster.
The group was added in [the common directory](https://github.com/operate-first/apps/blob/484ae6e797f5f1271ec8dc11ce277de5faadebf0/cluster-scope/overlays/moc/common/kustomization.yaml#L9) so it will be deployed in all of the MOC clusters.

**Comment by tumido** [comment]
@paoloantinori Yup that's exactly the last step (+ some additional mentions during the doc). Looking at it now, it's not very understandable and clear what should be done. I'll try rephrasing it, thank you for pointing that out. :+1:

**Review by tumido — COMMENTED** [review]
/lgtm thank you! :+1:

**Review by 4n4nd — APPROVED** [review]

**Comment by tumido** [comment]
@paoloantinori may I ask you, can you please upload your public key to the GPG key you've used to encrypt the group resource (`4B6A81C7D1711864408480CEC37C8EDF9301474C`) to any of the GPG key servers mentioned [here](https://github.com/operate-first/apps/blob/master/cluster-scope/overlays/moc/README.md) (or any public one really). We need to be able to import your pubkey in case we would like to update the resource for you - we can't reencrypt it unless we hold your pubkey as well. Can you please share it? :slightly_smiling_face:

**Comment by paoloantinori** [comment]
Here you go: https://keys.openpgp.org/search?q=pantinor@redhat.com

**Comment by tumido** [comment]
Thank you sir! @paoloantinori  :+1: :rocket:
